src/Sylius/Bundle/ProductBundle/Form/Type/ProductAssociationsType.php

```diff
@@ -39,10 +46,11 @@ public function configureOptions(OptionsResolver $resolver): void
     {
         $resolver->setDefaults([
             'entries' => $this->productAssociationTypeRepository->findAll(),
-            'entry_type' => TextType::class,
             'entry_name' => fn (ProductAssociationTypeInterface $productAssociationType) => $productAssociationType->getCode(),
+            'entry_type' => ProductAutocompleteChoiceType::class,
```

Comment:
The form type in ProductBundle should probably stay as it is and you should extend the type in AdminBundle as you use here a type which exists only in AdminBundle.